### DISCUSSION TITLE
WiscKey 论文阅读 | Wu's Blog

### DISCUSSION THREAD

**Discussion by giscus** [gogim1/gogim1.github.io#5] [General]
# WiscKey 论文阅读 | Wu's Blog

https://gogim1.github.io/posts/wisckey

**Comment by gogim1** [comment]
适用场景：大 Value、写多读少、点查多范围查询少

**Comment by gogim1** [comment]
> 相比leveldb，WiscKey不支持快照机制，因为垃圾回收机制会重新将有效的数据put到db中，这就导致了原来在lsm树深层的记录重新回到了lsm浅层的位置，我们知道leveldb读的顺序就是由浅往深。

？如何理解

> 拿vLog当WAL，岂不是和vLog的buffer pool冲突了？ 当然是，要是vLog有了buffer pool，就不能保证vLog的崩溃一致性了，所以就不能当作WAL，实际上，现在的一些基于KV分离的LSM Tree的存储引擎，都保留了WAL

有道理

**Reply by gogim1** [reply]
1.  [垃圾回收的实现](https://www.skyzh.dev/posts/articles/2021-08-07-lsm-kv-separation-overview/#%E5%9E%83%E5%9C%BE%E5%9B%9E%E6%94%B6%E7%9A%84%E5%AE%9E%E7%8E%B0)
